app/javascript/packages/password-strength/password-strength-element.ts

```diff
@@ -160,11 +160,23 @@ class PasswordStrengthElement extends HTMLElement {
       const result = zxcvbn(this.input.value, this.forbiddenPasswords);
       const score = this.#getNormalizedScore(result);
       this.setAttribute('score', String(score));
+      const inputDescribedBy = this.input.getAttribute('aria-describedby');
+      if (!inputDescribedBy?.includes('password-strength')) {
+        this.input.setAttribute(
+          'aria-describedby',
+          ['password-strength', inputDescribedBy].join(' '),
+        );
+      }
       this.input.setCustomValidity(
         this.#isValid(result) ? '' : t('errors.messages.stronger_password'),
       );
       this.strength.textContent = this.#getStrengthLabel(score);
       this.feedback.textContent = this.#getNormalizedFeedback(result);
+    } else {
+      this.input.setAttribute(
+        'aria-describedby',
+        this.input.getAttribute('aria-describedby')!.replace(/\s*password-strength\s*/, ''),
```

Comment:
I think this is a case where TypeScript is right in warning us that we can't guarantee that `aria-describedby` attribute is there. I might suggest using the [optional chaining operator (`?.`)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) instead of the non-null assertion (`!.`).

```suggestion
        this.input.getAttribute('aria-describedby')?.replace(/\s*password-strength\s*/, ''),
```

Reply:
> This works if I add a non-null to the end otherwise TS was still complaining.

Can you share the error you are seeing?

Reply:
```
Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.ts(2345)
```
![Screenshot 2025-02-25 at 9 20 23 AM (2)](https://github.com/user-attachments/assets/276dcd69-d12f-4206-8eca-d239a21763f4)

Reply:
Ah, it's because the optional chaining will result in trying to set `null` as the value if there is no attribute present. Maybe we could address this by changing the `else` to an `else if (this.input.hasAttribute('aria-describedby')`, and then it'd be safe to use the non-null assertion.

Reply:
Yeah, that seems like a sensible approach 👍 

I think the issue you were seeing is that TypeScript can't infer a connection between `hasAttribute` and `getAttribute` to know that the condition satisfied that the attribute definitely exists, whereas storing and reusing the variable reference, it knows it's safe to use if already checked in the condition.

Reply:
Thank you very much!